
# HG changeset patch
# User Mads Kiilerich <mads@kiilerich.com>
# Date 1728323604 -7200
# Node ID 141ed0bf72a1fe00f5a9ea105bb042a182c56fc5
# Parent  7b8769cca23dc78e9dd0f0eec52dcc467c76ddcb
hgdemandimport: add collections.abc to the ignore list because Python 3.13 rc3

It worked in Python 3.13 rc2, but something changed for rc3.

Now, when i18n.py imports typing and it touches collections.abc.Hashable , we
get:

  ValueError: module object for collections.abc substituted in sys.modules during a lazy load

This seems to be a general problem, released in the final Python 3.13 . I have
not analyzed the problem in details.

We *could* work around the problem by disabling demand import of typing. But
that would effectively disable demand import of collections.abc too. Instead,
just disable demand import for collections.abc .

diff -r 7b8769cca23d -r 141ed0bf72a1 hgdemandimport/__init__.py
--- a/hgdemandimport/__init__.py	Tue Oct 08 20:50:46 2024 +0200
+++ b/hgdemandimport/__init__.py	Mon Oct 07 19:53:24 2024 +0200
@@ -62,6 +62,7 @@
     '_weakrefset',
     'warnings',
     'threading',
+    'collections.abc',
 }
 
 _pypy = '__pypy__' in sys.builtin_module_names

